sequence/src/test/java/com/vitreoussoftware/bioinformatics/sequence/encoding/EncodingSchemeTestBase.java

```diff
@@ -67,4 +73,49 @@ public void theoryBasePairToStringFromLower(EncodingScheme scheme, Character cha
 
         assertThat(character.toString(), is(basePair.toString()));
     }
+
+    /**
+     * Given the set of valid pairs ensure we can complement between them in the 'forward' direction
+     */
+    @Theory
+    public void theoryBasePairComplement0to1(EncodingScheme scheme, Pair<Character, Character> pair) {
+        val basePair0 = scheme.fromCharacter(pair.getValue0());
+        val basePair1 = scheme.fromCharacter(pair.getValue1());
+
+        assertThat(basePair0.complement(), is(basePair1));
+    }
+
+    /**
+     * Given the set of valid pairs ensure we can complement between them in the 'backward' direction
+     */
+    @Theory
+    public void theoryBasePairComplement1to0(EncodingScheme scheme, Pair<Character, Character> pair) {
+        val basePair0 = scheme.fromCharacter(pair.getValue0());
+        val basePair1 = scheme.fromCharacter(pair.getValue1());
+
+        assertThat(basePair1.complement(), is(basePair0));
+    }
+
+    @Test(expected = InvalidDnaFormatException.class)
+    public void testToBasePairInvalidByte() {
+        val scheme = getEncodingScheme();
+        scheme.toBasePair((byte) -1);
+    }
+
+    @Test(expected = InvalidDnaFormatException.class)
+    public void testToCharInvalidByte() {
+        val scheme = getEncodingScheme();
+        scheme.toChar((byte) -1);
+    }
+
+    @Test
+    public void testBasePairComplementOtherEncodingScheme() {
+        val scheme = getEncodingScheme();
+        val otherScheme = getOtherEncodingScheme();
+
+        val bp = otherScheme.fromCharacter('A');
+        val expected = scheme.fromCharacter('T');
```

Comment:
All of the potential complements are managed via the theories above. This test is to ensure that when asking to complement something from a different encoding scheme (which might honestly be an error) will work.